src/blockchain-scanner.service.ts

```diff
@@ -90,9 +90,14 @@ export class BlockchainScannerService implements OnApplicationBootstrap {
         if (filteredEvents.length > 0) {
           this.logger.debug(`Found ${filteredEvents.length} delegations at block #${currentBlockNumber}`);
         }
-        const jobs = filteredEvents.map(({ event }) => {
+        const jobs = filteredEvents.map(async (event) => {
           const { key: jobId, data } = createGraphUpdateJob(event.data.delegatorId, event.data.providerId, UpdateTransitiveGraphs);
-          return this.graphUpdateQueue.add(`graphUpdate:${event.data.delegatorId}`, data, { jobId });
+          const job = await this.graphUpdateQueue.getJob(jobId);
+          if (job) {
+            await job.retry();
```

Comment:
if job already exists, re-try it

Reply:
Otherwise stuff doesn't happen in the order you expect...

Reply:
Maybe better to create an `enqueueJob` async method that encapsulates this logic & just returns a Promise, then the main structure of the code here can be pretty much as before, just replacing `queue.add` with `this.enqueueJob`

Reply:
its awaiting in next line        ` await Promise.all(jobs);`